volunteer.slider.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/**
+ * Delegated implementation of hook_civicrm_buildForm
+ *
+ * Customizes the UI for adding custom fields to allow the user to specify whether
+ * a multi-select field should use the slider widget or not
+ */
+function _volunteer_civicrm_buildForm_CRM_Custom_Form_Field($formName, CRM_Core_Form &$form) {
+  // set default value for the checkbox
+  $field_id = $form->getVar('_id');
+  $widgetized_fields = _volunteer_get_slider_fields();
+  $form->_defaultValues['is_slider_widget'] = in_array($field_id, $widgetized_fields);
+
+  // add checkbox to the form object
+  $form->add('checkbox', 'is_slider_widget', ts('Use Slider Widget?'));
+
+  // add checkbox to the display
+  CRM_Core_Region::instance('page-body')->add(array(
+   'template' => 'Slider/CRM/Custom/Form/Field.tpl',
+  ));
+
+  // reposition and show/hide checkbox
+  CRM_Core_Resources::singleton()->addScriptFile('org.civicrm.volunteer', 'js/CRM_Custom_Form_Field.js');
+}
+
+/**
+ * Delegated implementation of hook_civicrm_postProcess
+ *
+ * Handles the "Use Slider Widget?" field added to the custom fields UI
+ */
+function _volunteer_civicrm_postProcess_CRM_Custom_Form_Field($formName, &$form) {
+  $is_slider_widget = CRM_Utils_Array::value('is_slider_widget', $form->_submitValues);
+  $custom_field_id = $form->getVar('_id');
+
+  $verb = $is_slider_widget ? CRM_Core_Action::ADD : CRM_Core_Action::DELETE;
+  _volunteer_update_slider_fields(array($verb => $custom_field_id));
+}
+
+/**
+ * Delegated implementation of hook_civicrm_buildForm
+ *
+ * Registers the form to allow use of the volunteer slider widget.
+ */
+function _volunteer_civicrm_buildForm_CRM_Profile_Form_Edit($formName, CRM_Core_Form &$form) {
+  $form->allowVolunteerSliderWidget = TRUE;
+}
+
+/**
+ * For forms which have registered as slider-enabled, add the JS and CSS necessary
+ * to render the slider widget(s).
+ *
+ * @param CRM_Core_Form $form
+ */
+function _volunteer_addSliderWidget(CRM_Core_Form &$form) {
+  if (isset($form->allowVolunteerSliderWidget) && $form->allowVolunteerSliderWidget) {
+    $db_widgetized_fields = _volunteer_get_slider_fields();
+    foreach ($db_widgetized_fields as &$value) {
+      $value = 'custom_' . $value;
+    }
+    $form_element_names = array_flip($form->_elementIndex);
```

Comment:
Add comment: Compare the global list of widget-enabled fields with the element in this form to get a list of widget-enabled fields in this form.